src/Illuminate/Foundation/Testing/DatabaseConnectionFactory.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Illuminate\Foundation\Testing;
+
+use Illuminate\Contracts\Container\Container;
+use Illuminate\Contracts\Database\Connectors\ConnectionFactory as ConnectionFactoryContract;
+use Illuminate\Database\Connectors\ConnectionFactory;
+
+/**
+ * @internal
+ */
+class DatabaseConnectionFactory extends ConnectionFactory
+{
+    /**
+     * List of cached database connections.
+     *
+     * @var array<string, \Illuminate\Database\Connection>
+     */
+    protected static array $cachedConnections = [];
+
+    /**
+     * Create a new connection factory instance.
+     *
+     * @param  \Illuminate\Contracts\Container\Container  $container
+     * @return void
+     */
+    public function __construct(
+        Container $container,
+        protected ConnectionFactoryContract $factory
+    ) {
+        parent::__construct($container);
+    }
+
+    /**
+     * Establish a PDO connection based on the configuration.
+     *
+     * @param  array  $config
+     * @param  string  $name
+     * @return \Illuminate\Database\Connection
+     */
+    #[\Override]
+    public function make(array $config, $name)
+    {
+        $key = $name ?? $config['name'];
+
+        // SQLite doesn't have any max connections limitation so it should be safe to just create a new connection between tests.
+        if ($config['driver'] === 'sqlite') {
+            return $this->factory->make($config, $name);
+        }
+
+        if (! isset(static::$cachedConnections[$key]) || is_null(static::$cachedConnections[$key]->getRawPdo() ?? null)) {
+            return static::$cachedConnections[$key] = $this->factory->make($config, $name);
+        }
+
+        $config = $this->parseConfig($config, $name);
+
+        $connection = $this->createConnection(
+            $config['driver'], static::$cachedConnections[$key]->getRawPdo(), $config['database'], $config['prefix'], $config
+        )->setReadPdo(static::$cachedConnections[$key]->getRawReadPdo());
+
+        return static::$cachedConnections[$key] = $connection;
```

Comment:
```suggestion
        static::$cachedConnections[$key]->disconnect();
        return static::$cachedConnections[$key] = $this->factory->make($config, $name);
```

[IMHO]
I'm concerned that reusing connections may have unintended side effects. How about trying disconnecting and reconnecting?

Reply:
The whole point of this PR is to reuse available PDO instances, this is inline with what's already approved to in-memory database feature available in `RefreshDatabase` for Laravel 11.

Reply:
There is a risk that the transaction state and session settings will be carried over as well, making it difficult to understand why the test failed.What do you think about this?

These are just the cases I can think of right now, so I may have missed something.

Reply:
Wouldn't that already been handled by `DatabaseTransactionManager`? Can you provide a use case where `DB::transaction()` is run without being reset to 0?

Reply:
If using two-phase commit, it can easily happen because Laravel does not control the transaction.

Also, the problem is more with the session settings, and when the transaction isolation level is changed, the level is reflected in the next test as well.

Reply:
So all this issue should also exist in Laravel Octane, right?

Reply:
Probably not. The PDO instance you get should be different each time since the DI container is reset each time.

(I haven't really looked into how 8.x handles database connections yet, so I could be wrong.)